source/utilities.tex

```diff
@@ -11806,30 +11806,54 @@
 are general-purpose memory resources having the following qualities:
 \begin{itemize}
 \item
-Each resource frees its allocated memory on destruction,
-even if \tcode{deallocate} has not been called for some of the allocated blocks.
+A pool resource has an \defn{upstream memory resource},
+supplied on construction, which is used to allocate memory that
+the pool resource subdivides to satisfy calls to its own
+\tcode{do_allocate} member.
+
+\item
+A \defn{chunk} is a region of storage allocated
+from the upstream memory resource. A pool resource divides
+each chunk into one or more blocks of the same block size.
+Different chunks can have different block sizes, and different
+numbers of blocks.
+
+\item
+A \defn{pool} is a collection of chunks that all have the same
+block size. Each pool resource contains a collection of pools
+with different block sizes.
+
+\item
+On destruction, a pool resource frees all the memory it obtained from
+its upstream memory resource by calling the upstream resource's
+\tcode{deallocate} function.
+\begin{note}
+This frees all memory returned by the pool resource's \tcode{do_allocate}
+function, even if \tcode{do_deallocate} has not been called for some
+of the allocated blocks.
+\end{note}
+
 \item
-A pool resource consists of a collection of \defn{pools},
-serving requests for different block sizes.
-Each individual pool manages a collection of \defn{chunks}
-that are in turn divided into blocks of uniform size,
-returned via calls to \tcode{do_allocate}.
 Each call to \tcode{do_allocate(size, alignment)} is dispatched
-to the pool serving the smallest blocks accommodating at least \tcode{size} bytes.
+to the pool serving the smallest blocks accommodating at \tcode{size} bytes
+and are aligned to a multiple of \tcode{alignment}.
```

Comment:
This sentence has gone astray: "... accommodating at `size` bytes and are aligned to a multiple of `alignment`." is not Englishing goodly. How about "Each call to `do_allocate(size, alignment)` is dispatched to the pool serving the smallest blocks that accommodate `size` and `alignment`."? "multiple of `alignment` seems extraneous to me given that all alignments are powers of two (http://eel.is/c++draft/basic.align#4).

Reply:
Yes I'm not sure how that got mangled. I'll push an update.